R/reactives.R

```diff
@@ -469,10 +473,15 @@ ReactiveValues <- R6Class(
 
     # Mark a value as frozen If accessed while frozen, a shiny.silent.error will
     # be thrown.
-    freeze = function(key) {
+    freeze = function(key, invalidate = FALSE) {
       domain <- getDefaultReactiveDomain()
       rLog$freezeReactiveKey(.reactId, key, domain)
       setMeta(key, "frozen", TRUE)
+
+      if (invalidate) {
+        # Force an invalidation
+        self$set(key, NULL, force = TRUE)
```

Comment:
Setting it to `NULL` feels a little weird to me. Is there any way this value could leak out?

Reply:
Discussed in person; even if it could leak out, it makes at least as much sense for it to leak as `NULL` versus leaking as the last-known-good value.